R/teal_data-get_code.R

```diff
@@ -6,9 +6,57 @@
 #' Use `datanames` to limit the code to one or more of the data sets enumerated in `@datanames`.
 #' If the code has not passed verification (with [`verify()`]), a warning will be prepended.
 #'
-#' @section Notes for Developers:
-#' To learn more about how a subset of code needed to reproduce a specific data set is extracted from all code,
-#' see [`get_code_dependency()`].
+#' @section Extracting dataset-specific code:
+#' When `datanames` is specified, the code returned will be limited to the code needed to create the
+#' requested data sets. `code` stored in the `teal_data` object is statically analyzed to determine
+#' dependency tree between each line of the code. Analysis is performed automatically basing on the
+#' used symbols and it is working in a typical case when new dataset is created by assignment operator.
+#' For example:
+#' ```r
+#' data <- teal_data() |>
+#'   eval_code("
```

Comment:
@chlebowa I was thinking to use `within` in these examples and use `eval_code` only when `# @linksto` is added (with the note that `within` discards comments)

Reply:
+1